core/src/main/scala/org/ensime/indexer/IndexService.scala

```diff
@@ -30,7 +30,7 @@ object IndexService extends SLF4JLogging {
       def spacey(s: String) = List(s, s.replace('.', ' '))
 
       val shortPkg = path.map(_.charAt(0)).mkString("", ".", "." + name)
-      val innerNames = name.split('$').filter(_.size > 3).flatMap(cases)
+      val innerNames = name.split('$').flatMap(cases)
```

Comment:
This is what was causing problems with @ChrisCoffey PR, I'm not entirely sure why `filter(_.size > 3)` is ever needed here. Perhaps @fommil could explain this.